src/cmd/services/m3aggregator/main/main.go

```diff
@@ -83,19 +86,46 @@ func main() {
 		SetMetricsSamplingRate(cfg.Metrics.SampleRate()).
 		SetReportInterval(cfg.Metrics.ReportInterval())
 
-	// Create the raw TCP server options.
-	rawTCPAddr := cfg.RawTCP.ListenAddress
-	rawTCPServerScope := scope.SubScope("rawtcp-server").Tagged(map[string]string{"server": "rawtcp"})
-	iOpts := instrumentOpts.SetMetricsScope(rawTCPServerScope)
-	rawTCPServerOpts := cfg.RawTCP.NewServerOptions(iOpts)
+	var (
+		m3msgAddr        string
+		m3msgServerOpts  m3msg.Options
+		rawTCPAddr       string
+		rawTCPServerOpts rawtcp.Options
+		httpAddr         string
+		httpServerOpts   http.Options
+	)
+	if cfg.M3Msg != nil {
+		// Create the M3Msg server options.
+		m3msgAddr = cfg.M3Msg.Server.ListenAddress
+		m3msgInsrumentOpts := instrumentOpts.
+			SetMetricsScope(scope.
+				SubScope("m3msg-server").
```

Comment:
nit: Do we need both subscope and tags?

Reply:
Unfortunately I have to preserve the "rawtcp-server" metrics which had both subscope and tags, so will just keep it consistent across both of them for now.